### DISCUSSION TITLE
Merge Swoole and OpenSwoole to make unified Framework / Repo

### DISCUSSION THREAD

**Discussion by fakharksa** [swoole/swoole-src#4914] [Ideas]
@deminy 
I strongly suggest that two separated Repos be merged into one, and both communities should work together by sharing / respecting new ideas. This separation of Repos/ Community is causing a lot of dent / hinderance in the use / recommendation of this great technology.

For Example, read this Tweet from Taylor Otwell (Founder of Laravel)
[Taylor Otwell Tweet on Swoole](https://twitter.com/taylorotwell/status/1463208951076503582?s=20&t=DUv-BITNDvJNjZt4UySgmA)

If there is some difference in implementation, then that should be resolved with OO-pattern like Factory Method, Adapter / Decorator (plug-in Design) so as to leave it on Application Developers to choose from one of the implementations / ideas.

This will resolve issues with marketing the framework; the communities of which are on the same core Objectives / Agenda. (Providing Coroutine based Asynchrony and Multi-processing around different Network Protocols, and I/O Multiplexing in PHP etc.).

I think the name of the merged Repo should still be OpenSwoole as the prefix "Open" serves best in conveying the idea that Swoole is OpenSource and free to use under OpenSource License.

I hope that the two communities (Swoole and OpenSwoole) arrange a meeting and sort out how to go with re-merging into one unified Repo / Framework.

I have put up same request on OpenSwoole GitHub too.
[Request on OpenSwoole GitHub](https://github.com/openswoole/swoole-src/discussions/275#discussion-4600490)

**Comment by Yurunsoft** [comment]
OpenSwoole is forked from Swoole.

As a user, I feel that Swoole has been very open.

Issues and PRs are handled quickly, which is one of the reasons why I chose to use Swoole in the first place.

Also, for this discussion maybe you should @matyhtf .

For the name, I think you might also need OpenLinux, OpenPHP, OpenMySQL, OpenNginx ......

**Comment by ClosetGeek-Git** [comment]
There was an ideological difference between some in the Swoole community which lead to the openswoole fork. I don't see the two merging unless those differences have been resolved. It was a thorny issue as well, so understand that some may choose to ignore this question completely.

**Comment by ClosetGeek-Git** [comment]
BTW, most people don't look to name to see how "open" a project is, but to the license. Swoole is Apache 2. And as for how open the team is I can vouch for them because of this issue here https://github.com/matyhtf/phpx/issues/63 which was in 2019.

**Comment by leocavalcante** [comment]
I think the original project and the fork will remain as separate projects for last long as many others around the software industry world. Unfortunately, the fork chose a misleading name.

**Comment by deba12** [comment]
I think there is also license / copyright issue with openswoole,
you can look in the openswoole source code there are license changes in all file headers and removal of copyright info 

just my two cents.
